src/app/home-page/recent-events/recent-events.component.ts

```diff
@@ -31,15 +39,38 @@ export class RecentEventsComponent implements OnInit {
     Event.TypeEnum.PUBLISHENTRY,
     Event.TypeEnum.UNPUBLISHENTRY,
   ];
-  constructor(private recentEventsQuery: RecentEventsQuery, private recentEventsService: RecentEventsService) {}
+  constructor(
+    private recentEventsQuery: RecentEventsQuery,
+    private recentEventsService: RecentEventsService,
+    private activatedRoute: ActivatedRoute,
+    private usersService: UsersService,
+    private alertService: AlertService,
+    protected userQuery: UserQuery
+  ) {
+    this.username = this.activatedRoute.snapshot.paramMap.get('username');
+  }
 
   ngOnInit() {
+    if (this.username) {
+      this.usersService.listUser(this.username).subscribe(
+        (currentUser: User) => {
+          this.recentEventsService.get(currentUser);
+        },
+        (error: HttpErrorResponse) => {
+          this.alertService.detailedError(error);
+        }
+      );
+    } else {
+      this.userQuery.user$.subscribe((user: User) => {
+        this.recentEventsService.get(user);
```

Comment:
You may need to check that `user` is defined, but I'm not sure. It could be related to the question you asked about seeing an `undefined` error (but just guessing).